src/main/java/it/gov/pagopa/admissibility/service/build/BeneficiaryRule2DroolsRuleImpl.java

```diff
@@ -117,15 +107,8 @@ private ExtraFilter automatedCriteria2ExtraFilter(AutomatedCriteriaDTO automated
         return new NotOperation(new Filter(field, operator, automatedCriteriaDTO.getValue()));
     }
 
-    private Map<String, LocalDate> setStartAndEndDate(Initiative2BuildDTO initiative){
-        Map<String, LocalDate> startAndEndDates = new HashMap<>();
-        if(initiative.getGeneral().getRankingStartDate()==null || initiative.getGeneral().getRankingEndDate()==null) {
-            startAndEndDates.put(START,initiative.getGeneral().getStartDate());
-            startAndEndDates.put(END,initiative.getGeneral().getEndDate());
-        }else{
-            startAndEndDates.put(START,initiative.getGeneral().getRankingStartDate());
-            startAndEndDates.put(END,initiative.getGeneral().getRankingEndDate());
-        }
-        return startAndEndDates;
+    private void setStartAndEndDate(Initiative2BuildDTO initiative, InitiativeConfig initiativeConfig){
```

Comment:
as the setter thanks to firstNonNull became simple, it's not more necessary to create a separate function to perform the setters! put them directly on the metho above